packages/common/src/components/Filter/GroupedEnumFilter.tsx

```diff
@@ -68,64 +71,55 @@ export const GroupedEnumFilter = ({
     onSelectedEnumIdsChange([...selectedEnumIds.filter((id) => id2enum[id]), id]);
   };
 
-  // put the IDs needed for compareTo (although not part of the interface)
-  const toSelectOption = ({ id, groupId, label }): SelectOptionObject =>
-    ({
-      toString: () => label,
-      id,
-      groupId,
-      compareTo: (option) => option.id === id && option.groupId === groupId,
-    } as SelectOptionObject);
-
-  const options = supportedGroups.map(({ label, groupId }) => (
-    <SelectGroup key={groupId} label={label}>
-      {supportedEnumValues
-        .filter((item) => item.groupId === groupId)
-        .map(({ id, label }) => (
-          <SelectOption key={id} value={toSelectOption({ id, label, groupId })} />
-        ))}
-    </SelectGroup>
-  ));
-
-  const onFilter: (
-    event: React.ChangeEvent<HTMLInputElement> | null,
-    textInput: string,
-  ) => React.ReactElement[] | undefined = (_event, textInput) => {
-    if (textInput === '') {
-      return options;
-    }
-
-    const filteredGroups = options
-      .map((group) => {
-        const filteredGroup = React.cloneElement(group, {
-          children: group.props.children.filter((item) => {
-            // options are not plain strings but the our toString() method
-            // converts them in a meaningful way
-            return item.props.value.toString().toLowerCase().includes(textInput.toLowerCase());
-          }),
-        });
-        if (filteredGroup.props.children.length > 0) return filteredGroup;
-      })
-      .filter((newGroup) => newGroup);
-    return filteredGroups;
-  };
-
-  const onSelect: (
-    event: SelectEventType,
-    value: SelectValueType,
-    isPlaceholder?: boolean,
-  ) => void = (_event, value, isPlaceholder) => {
-    if (isPlaceholder) {
-      return;
-    }
-    const id = typeof value === 'string' ? value : (value as EnumValue).id;
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | number | undefined,
+  ) => {
+    const label = typeof value === 'string' ? value : value?.toString();
```

Comment:
u think this is neccessry? for example IIRC also string has toString() method so no need to check for === string - please check.